lasagne/layers/special.py

```diff
@@ -354,3 +355,95 @@ def _meshgrid(height, width):
+        numpy array is provided the shape should  match the `incoming` shape
+        in all dimension provided by the `axes` parameter and 1 in all others.
+        See :meth:`Layer.create_param` for more information.
+
+    axes : 'auto', 'all', int or tuple of int
+        The axes along which the parameters of the rectifier units are
+        going to be shared. If ``'auto'`` (the default),
+        share over all axes except for the second: this will share biases over
+        the minibatch dimension for dense layers, and additionally over all
+        spatial dimensions for convolutional layers. If ``'all'`` share over
+        all axes which corresponds to a single scalar parameter
+
+    **kwargs
+        Any additional keyword arguments are passed to the `Layer` superclass.
+
+     References
+    ----------
+    .. [1] K He, X Zhang et al. (2015):
+       Delving Deep into Rectifiers: Surpassing Human-Level Performance on
+       ImageNet Classification,
+       http://link.springer.com/chapter/10.1007/3-540-49430-8_2,
+    """
+    def __init__(self, incoming, alpha=init.Constant(0.25), axes='auto',
+                 **kwargs):
+        super(ParametricRectifierLayer, self).__init__(incoming, **kwargs)
+        if axes == 'auto':
+            self.axes = (0,) + tuple(range(2, len(self.input_shape)))
+        elif axes == 'all':
+            self.axes = tuple(range(len(self.input_shape)))
+        elif isinstance(axes, int):
+            self.axes = (axes,)
+        else:
+            self.axes = axes
+
+        shape = [size for axis, size in enumerate(self.input_shape)
+                 if axis not in self.axes]
+        if any(size is None for size in shape):
+            raise ValueError("ParametricRectifierLayer needs specified input "
+                             "sizes for all dimensions/axes which constitute "
+                             "the same parameters.")
+        self.alpha = self.add_param(alpha, shape, name="alpha",
+                                    regularizable=False)
+
+    def get_output_for(self, input, **kwargs):
+        axes = iter(range(self.alpha.ndim))
+        pattern = ['x' if input_axis in self.axes
+                   else next(axes)
+                   for input_axis in range(input.ndim)]
+        alpha = self.alpha.dimshuffle(pattern)
+        return theano.tensor.nnet.relu(input, alpha)
+
+
+def parametric_relu(layer, **kwargs):
+    """
+    Convenience function to apply parametric rectify to a given layer's output.
+    Will set the layer's nonlinearity to identity if there is one and will
+    apply the parametric rectifier instead.
+
+    Parameters
+    ----------
+    layer: a :class:`Layer` instance
+        The `Layer` instance to apply the parametric rectifier layer to;
+        note that it will be irreversibly modified as specified above
+
+    **kwargs
+        Any additional keyword arguments are passed to the
+        `ParametricRectifierLayer` class.
```

Comment:
The correct markup is:
```
... to the :class:`ParametericRectifierLayer` class.
```
Otherwise it won't be linked to it in the generated documentation.